src/components/calcite-card/calcite-card.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  Prop
+} from "@stencil/core";
+import { CSS, SLOTS } from "./resources";
+import { ENTER, SPACE } from "../../utils/keys";
+import { getElementDir } from "../../utils/dom";
+import { VNode } from "@stencil/core/dist/declarations";
+
+/**
+ * @slot thumbnail - [Required] A slot for adding a thumnail to the card.
+ * @slot header - [Required] A slot for adding a heading and an icon to the card.
+ * @slot - A slot for adding subheader/description content.
+ * @slot action - A slot for adding a single action as a button.
+ * @slot footer-leading - A slot for adding a leading footer.
+ * @slot footer-trailing - A slot for adding a trailing footer.
+ */
+
+@Component({
+  tag: "calcite-card",
+  styleUrl: "calcite-card.scss",
+  shadow: true
+})
+export class CalciteCard {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element() el: HTMLCalciteCardElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Public Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**  true, the card can't be clicked and is visually muted.  */
+  @Prop({ reflect: true }) disabled?: boolean = false;
+
+  /**  When true, the cards content is waiting to be loaded. This state shows a busy indicator.*/
+  @Prop({ reflect: true }) loading?: boolean = false;
+
+  /**
+   * Indicates whether the image's height is respected for the crop of the thumbnail.
+   * When false (default), the image fills the whole thumbnail space.
+   * When true, the height of the thumbnail is used to crop and may not fill the container for the thumbnail.
+   */
+  @Prop({ reflect: true }) respectImageHeight?: boolean = false;
+
+  /** Indicates whether the card is selected. */
+  @Prop({ reflect: true, mutable: true }) selected?: boolean = false;
+
+  /** Indicates whether the card is selected. */
+  @Prop({ reflect: true, mutable: true }) selectable?: boolean = false;
+
+  /**  The theme of the card.*/
+  @Prop({ reflect: true, mutable: true }) theme: "light" | "dark" = "light";
+
+  /**  The theme of the card.*/
+  @Prop({ reflect: true, mutable: true }) appearance: "wide" | "default" =
+    "wide";
+
+  //--------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  //--------------------------------------------------------------------------
+
+  /** Fired when a selectable card is selected */
+  @Event() calciteCardSelected: EventEmitter;
```

Comment:
should this be `calciteCardSelect` and not past tense?

Reply:
Yeah and additionally I need to add a public method to allow cards to be pragmatically selected as well... 👷

Reply:
Wouldn't the selected attribute be enough?

Reply:
Shouldn't need a method. attribute > method :)